lib/Github/Api/Deployment/Policies.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Github\Api\Deployment;
+
+use Github\Api\AbstractApi;
+use Github\Exception\MissingArgumentException;
+
+/**
+ * Listing, creating and updating deployments.
+ *
+ * @link https://docs.github.com/en/rest/deployments/branch-policies?apiVersion=2022-11-28#list-deployment-branch-policies
+ */
+class Policies extends AbstractApi
+{
+    /**
+     * List deployment branch policies
+     *
+     * @link https://docs.github.com/en/rest/deployments/branch-policies?apiVersion=2022-11-28#list-deployment-branch-policies
+     *
+     * @param string $username   the username of the user who owns the repository
+     * @param string $repository the name of the repository
+     * @param string $environment the name of the environment.
+     * @param array  $params     query parameters to filter deployments by (see link)
+     *
+     * @return array the deployments requested
+     */
+    public function all(string $username, string $repository, string $environment, array $params = [])
+    {
+        return $this->get('/repos/'.rawurlencode($username).'/'.rawurlencode($repository).'/environments/'.rawurlencode($environment).'/deployment-branch-policies', $params);
+    }
+
+    /**
+     * Get a deployment branch policy
+     * 
+     * @link https://docs.github.com/en/rest/deployments/branch-policies?apiVersion=2022-11-28#get-a-deployment-branch-policy
+     *
+     * @param string $username   the username of the user who owns the repository
+     * @param string $repository the name of the repository
+     * @param string $environment the name of the environment.
+     * @param int    $id         The unique identifier of the branch policy.
+     *
+     * @return array
+     */
+    public function show(string $username, string $repository, string $environment, int $id)
+    {
+        return $this->get('/repos/'.rawurlencode($username).'/'.rawurlencode($repository).'/environments/'.rawurlencode($environment).'/deployment-branch-policies/'.$id);
+    }
+
+    /**
+     * Creates a deployment branch policy for an environment.
+     *
+     * @link https://docs.github.com/en/rest/deployments/branch-policies?apiVersion=2022-11-28#create-a-deployment-branch-policy
+     *
+     * @param string $username   the username of the user who owns the repository
+     * @param string $repository the name of the repository
+     * @param string $environment the name of the environment.
+     *
+     * @throws MissingArgumentException
+     *
+     * @return array information about the deployment branch policy
+     */
+    public function create(string $username, string $repository, string $environment, array $params)
+    {
+        if (!isset($params['name'])) {
+            throw new MissingArgumentException(['name']);
+        }
```

Comment:
I would not add this validation in our code, but let the github api validate the request and return any errors related to missing parameters. This way we won't block user in the future if github changes the required parameters